opendevin/controller/agent_controller.py

```diff
@@ -185,6 +185,9 @@ async def on_event(self, event: Event):
             elif isinstance(event, AgentDelegateObservation):
                 await self.add_history(NullAction(), event)
                 logger.info(event, extra={'msg_type': 'OBSERVATION'})
+            elif isinstance(event, ErrorObservation):
+                await self.add_history(NullAction(), event)
+                logger.info(event, extra={'msg_type': 'OBSERVATION'})
 
```

Comment:
How about adding the else part too and filtering unwanted events?

Reply:
Please correct me if I misunderstood, but I think what you are suggesting here is, doing 

```python
await self.add_history(NullAction(), event)
logger.info(event, extra={'msg_type': 'OBSERVATION'})
```

for every type of observation by default, and use a block-list to skip events we don't want to handle like this.

If that's what you mean, my answer is, it makes sense, but it's safer to do it this way (allow-list), unless we know most observations should be added to history.